pyomo/gdp/plugins/gdp_var_mover.py

```diff
@@ -91,3 +91,18 @@ def _apply_to(self, instance, **kwds):
                     Constraint, descend_into=Block, active=True)
                 for con in cons_in_disjunct:
                     con.deactivate()
+
+    def _disjunct_on_active_block(self, disjunct):
+        # Check first to make sure that the disjunct is not a
+        # descendent of an inactive Block or fixed and deactivated
+        # Disjunct, before raising a warning.
+        parent_block = disjunct.parent_block()
+        while parent_block is not None:
+            if parent_block.type() is Block and not parent_block.active:
+                return False
+            elif (parent_block.type() is Disjunct and not parent_block.active
+                  and parent_block.indicator_var.value == 0):
```

Comment:
I think you also need to test that `not parent_block.indicator_var.active`.  The block could have been deactivated by a transformation, and the indicator_var could have a current (initial) value of 0, but the `Disjunct` is still technically "active".

Reply:
Ah, I see your point. There really needs to be a better way of indicating that a disjunct has been relaxed. I think that I should check:
```
if parent_block.type() is Block and not parent_block.active and not hasattr(parent_block, 'indicator_var'):
    return False
elif (parent_block.type() is Disjunct and not parent_block.active
       and parent_block.indicator_var.value == 0 and parent_block.indicator_var.fixed):
```
Does that seem right?

Reply:
Should we be using that in the reclassify transformation, then?

Reply:
Note to self: @jsiirola said yes to this question walking by.